transports/quic/src/muxer.rs

```diff
@@ -0,0 +1,360 @@
+                        event
+                    );
+                }
+                ConnectionEvent::ConnectionLost(err) => return Poll::Ready(Err(err)),
+                ConnectionEvent::StreamOpened => {
+                    if let Some(waker) = inner.poll_outbound_waker.take() {
+                        waker.wake();
+                    }
+                }
+                ConnectionEvent::StreamReadable(substream) => {
+                    if let Some(substream) = inner.substreams.get_mut(&substream) {
+                        if let Some(waker) = substream.read_waker.take() {
+                            waker.wake();
+                        }
+                    }
+                }
+                ConnectionEvent::StreamWritable(substream) => {
+                    if let Some(substream) = inner.substreams.get_mut(&substream) {
+                        if let Some(waker) = substream.write_waker.take() {
+                            waker.wake();
+                        }
+                    }
+                }
+                ConnectionEvent::StreamFinished(substream)
+                | ConnectionEvent::StreamStopped(substream) => {
+                    if let Some(substream) = inner.substreams.get_mut(&substream) {
+                        if let Some(waker) = substream.read_waker.take() {
+                            waker.wake();
+                        }
+                        if let Some(waker) = substream.write_waker.take() {
+                            waker.wake();
+                        }
+                        if let Some(waker) = substream.finished_waker.take() {
+                            waker.wake();
+                        }
+                    }
+                }
+                ConnectionEvent::StreamAvailable => {
+                    if let Some(waker) = inner.poll_inbound_waker.take() {
+                        waker.wake();
+                    }
+                }
+            }
+        }
+        inner.poll_connection_waker = Some(cx.waker().clone());
+
+        // TODO: If connection migration is enabled (currently disabled) address
+        // change on the connection needs to be handled.
+
+        Poll::Pending
+    }
+
+    fn poll_inbound(
+        self: Pin<&mut Self>,
+        cx: &mut Context<'_>,
+    ) -> Poll<Result<Self::Substream, Self::Error>> {
+        let mut inner = self.inner.lock();
+        let substream_id = match inner.connection.accept_substream() {
+            Some(id) => {
+                inner.poll_outbound_waker = None;
+                id
+            }
+            None => {
+                inner.poll_inbound_waker = Some(cx.waker().clone());
+                return Poll::Pending;
+            }
+        };
+        inner.substreams.insert(substream_id, Default::default());
+        let substream = Substream::new(substream_id, self.inner.clone());
+        Poll::Ready(Ok(substream))
```

Comment:
I am a fan of whitespace, YMMV :upside_down_face: 

```suggestion
        let mut inner = self.inner.lock();
        
        let substream_id = match inner.connection.accept_substream() {
            Some(id) => {
                inner.poll_outbound_waker = None;
                id
            }
            None => {
                inner.poll_inbound_waker = Some(cx.waker().clone());
                return Poll::Pending;
            }
        };
        inner.substreams.insert(substream_id, Default::default());
        let substream = Substream::new(substream_id, self.inner.clone());
        
        Poll::Ready(Ok(substream))
```